lib/miq_automation_engine/engine/miq_ae_service/miq_ae_service_rbac.rb

```diff
@@ -0,0 +1,37 @@
+module MiqAeMethodService
+  module MiqAeServiceRbac
+    extend ActiveSupport::Concern
+
+    module ClassMethods
+      def find_ar_object_by_id(id)
+        rbac = Thread.current.thread_variable_get(:miq_rbac)
+        rbac ? Rbac.search(:class => model, :targets => [id], :results_format => :objects).first.first : model.send(:find, *id)
+      end
+
+      def all
+        rbac = Thread.current.thread_variable_get(:miq_rbac)
+        objs = rbac ? Rbac.search(:class => model, :results_format => :objects).first : model.send(:all)
+        wrap_results(objs)
+      end
+
+      def count
+        rbac = Thread.current.thread_variable_get(:miq_rbac)
+        rbac ? Rbac.search(:class => model).first.count : model.send(:count)
```

Comment:
==> `Rbac.filtered(model.all).count`

though, introducing `Rbac.count` is tempting